packages/react-components/src/components/RichTextEditor/Plugins/UpdateContentPlugin.tsx

```diff
@@ -62,7 +64,25 @@ export class UpdateContentPlugin implements EditorPlugin {
         break;
 
       case PluginEventType.ContentChanged:
-        this.onUpdate(UpdateEvent.ContentChanged);
+        if (
+          event.source.toLowerCase() === 'cut' ||
+          (event.source.toLowerCase() === 'keyboard' && (event.data === 8 || event.data === 46))
+        ) {
+          imageSrcArray = [];
+          event.contentModel?.blocks.map((block) => {
+            if (block.blockType === 'Paragraph') {
+              const segments = block.segments;
+              segments.map((segment) => {
+                if (segment.segmentType === 'Image') {
+                  imageSrcArray?.push(segment.src);
+                }
+              });
+            }
+          });
```

Comment:
Should we set `imageSrcArray` to `undefined` if nothing is pushed as result of
```
event.contentModel?.blocks.map((block) => {
            if (block.blockType === 'Paragraph') {
              const segments = block.segments;
              segments.map((segment) => {
                if (segment.segmentType === 'Image') {
                  imageSrcArray?.push(segment.src);
                }
              });
            }
          });
```
(`imageSrcArray` is equal [] after the code abov is executed)